### PR TITLE
New git datasource

### PR DESCRIPTION
**Pull request by hairyhenderson**
Adds support for accessing datasources in git repos. Adds 5 new schemes:

- `git://` - the git server (as served by `git daemon` and others)
- `git+file://` - git repo on the local filesystem - supports bare repos as well
- `git+ssh://` - git over ssh
- `git+http://`, `git+https://` - git over http(s)

`//` is used to separate repo from path, like `git+https://github.com/example/repo//path/to/file.yaml` - the path here is `/path/to/file.yaml`. Directories are supported. 

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

### CODE REVIEW

File: data/datasource_git.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+		u, err = url.Parse(arg)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse arg %s: %w", arg, err)
+	}
+	return u, err
+}
+
+func (g gitsource) parseQuery(orig *url.URL, arg *url.URL) string {
```
[the remark this answers is not in the document]
Comment (hairyhenderson) [reply data/datasource_git.go]: Thanks, @fixmie! 😂

File: data/datasource_git.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+		u.Path = "/" + u.Path
+	} else {
+		u, err = url.Parse(arg)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse arg %s: %w", arg, err)
+	}
+	return u, err
+}
```
Comment (rossedman) [inline data/datasource_git.go]: This is 👍

**Comment by rossedman** [comment]
@hairyhenderson This looks really awesome, thanks for throwing this up. I like where it's heading.

**Comment by hairyhenderson** [comment]
Thanks @rossedman! 😄

**Pull request merged**